opentelemetry/src/baggage.rs

```diff
@@ -146,10 +147,38 @@ impl Baggage {
         S: Into<BaggageMetadata>,
     {
         let (key, value, metadata) = (key.into(), value.into(), metadata.into());
-        if self.insertable(&key, &value, &metadata) {
-            self.inner.insert(key, (value, metadata))
-        } else {
-            None
+        if !key.as_str().is_ascii() {
+            return None;
+        }
+        let entry_content_len =
+            key_value_metadata_bytes_size(key.as_str(), value.as_str().as_ref(), metadata.as_str());
```

Comment:
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/src/baggage.rs#L59

Interesting..We accept Value for Baggage value, but the [spec only allows](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md#overview) strings!
**_Baggage values are any valid UTF-8 strings_**

@fraillt should we fix the API to accept OTelString only instead of Value?